addons/account/models/account_invoice.py

```diff
@@ -1364,7 +1365,8 @@ def _check_lines(self):
     @api.one
     def compute(self, value, date_ref=False):
         date_ref = date_ref or fields.Date.today()
-        amount = value
+        amount = abs(value)
+        sign = value < 0 and -1 or 1
```

Comment:
since the problem only occurs for line.value == 'fixed', I'd go for a slightly shorter patch of compute()

```diff
-                amt = round(line.value_amount, prec)
+                sign = value < 0 and -1 or 1
+                amt = sign * round(line.value_amount, prec)
```